pallets/services-payment/src/lib.rs

```diff
@@ -211,6 +216,29 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        pub fn give_free_credits(para_id: &ParaId) {
+            if GivenFreeCredits::<T>::contains_key(para_id) {
+                // This para id has already received free credits
+                return;
```

Comment:
We likely need to account weight for this, so we should at least return the weight used

Reply:
Done but I set the weight to Weight::default because it is not accounted for anywhere yet

Reply:
No, it is part of the mark_valid_for_collating extrinsic

Reply:
then we probably should find the way to benchmark it and account in such extrinsic. Probably through a trait?

Reply:
When we run the benchmarking script it is already taken into account.